src/ds/art.rs

```diff
@@ -0,0 +1,79 @@
+/// An adaptive radix tree.
```

Comment:
this is an experiment in reducing pointer density in the radix tree underlying many of our structures, not currently in use anywhere yet, barely started, mostly incomplete!